stores/user.ts

```diff
@@ -33,18 +34,27 @@ export const useUserStore = defineStore('user', () => {
   }
 
   async function updateBookUserProfile (payload: any) {
-    const { error } = await useFetch(`${LIKE_CO_API}/likernft/book/user/profile`, {
-      method: 'POST',
-      body: payload,
-      headers: {
-        authorization: `Bearer ${token.value}`
+    try {
+      isUpdatingBookUserProfile.value = true
+      const { error } = await useFetch(`${LIKE_CO_API}/likernft/book/user/profile`, {
+        method: 'POST',
+        body: payload,
+        headers: {
+          authorization: `Bearer ${token.value}`
+        }
+      })
+      if (error.value) {
+        throw error.value
       }
-    })
-    if (error.value) {
-      throw error.value
+      bookUser.value = { ...bookUser.value, ...payload }
+      return bookUser.value
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
+      throw error
```

Comment:
I think it's easier to read if we `console.error` on where we do need to finally catch the error though, instead of catching and throwing again